### PR TITLE
Implement 'borrowTrusted' and 'repayBorrowBehalfTrusted' functions on a market.

### PR DESCRIPTION
**Pull request by IgorSenychCW**
# Description

Added two new functions to the market contract:
- `borrowTrusted` - trusted borrower borrows assets from the protocol to their address
- `repayBorrowBehalfTrusted` - trusted repayer repays a borrow belonging to a trusted borrower

Updated market functions:
-  `borrow` - trusted borrowers disallowed to execute this function 
-  `repayBorrowBehalf` - will not be used for trusted repayments anymore

## Type of change
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

### CODE REVIEW

File: contracts/CTokenInterfaces.sol [hunk 1]
```diff
@@ -311,6 +311,11 @@ contract CErc20Interface is CErc20Storage {
     /*** Admin Functions ***/
 
     function _addReserves(uint addAmount) external returns (uint);
+
+    /*** Trusted Accounts ***/
+
+    function borrowTrusted(uint totalAmount, uint borrowAmount, address treasury) external returns (uint);
```
Comment (EvgeniiZaitsevCW) [inline contracts/CTokenInterfaces.sol:317]: The function has not clear name and names of its parameters.

The suggested version without changing of the function logic:
```
function borrowAndTransferTrusted(uint borrowAmount, uint nonTransferAmount, address to) external returns (uint);
```
where:
`borrowAmount` -- the amount that will be borrow from the market;
`nonTransferAmount` -- the amount that will be keep by the borrower himself from the borrowed amount, the other part will be sent to address `to`

The suggested version with changing of the function logic:
```
function borrowAndTransferTrusted(uint borrowAmount, uint transferAmount, address to) external returns (uint);
```
where:
`borrowAmount` -- the amount that will be borrow from the market;
`transferAmount` -- the amount that will be sent to address `to` from the borrowed amount, the other part will be kept by the borrower himself.

*A note:* If you agree with this remark, please do not forget to change the related code in other files too.

File: contracts/CTokenInterfaces.sol [hunk 2]
```diff
@@ -311,6 +311,11 @@ contract CErc20Interface is CErc20Storage {
...
+    function repayBorrowBehalfTrusted(address borrower, uint repayAmount) external returns (uint);
```
Comment (EvgeniiZaitsevCW) [inline contracts/CTokenInterfaces.sol:318]: The name does not reveal the full functionality.
The suggested version:
```
    function mintAndRepayBorrowBehalfTrusted(address borrower, uint repayAmount) external returns (uint);
```
A note: If you agree with this remark, please do not forget to change the related code in other files too.

File: contracts/CToken.sol [hunk 3]
```diff
@@ -1485,4 +1500,67 @@ contract CToken is CTokenInterface, Exponential, TokenErrorReporter {
...
+
+    /**
+      * @notice Trusted sender borrows assets from the protocol to their own address
+      * @param totalAmount The amount of the underlying asset to borrow + treasury amount
+      * @param borrowAmount The amount of the underlying asset to borrow
+      * @param treasury The treasury address
+      * @return uint 0=success, otherwise a failure (see ErrorReporter.sol for details)
+      */
+    function borrowTrustedInternal(uint totalAmount, uint borrowAmount, address treasury) internal nonReentrant returns (uint) {
+        uint error1 = accrueInterest();
```
Comment (EvgeniiZaitsevCW) [inline contracts/CToken.sol:1512]: A better name for the `error1` variable is `accrueResult`

**Review by EvgeniiZaitsevCW — CHANGES_REQUESTED** [review]
Only the changed code has been reviewed

File: contracts/CToken.sol (hunk 3 above)
Comment (IgorSenychCW) [reply contracts/CToken.sol]: Made naming changes: `error1` -> `accrueError` and `error2` -> `borrowError`.

**Review by EvgeniiZaitsevCW — APPROVED** [review]

**Pull request merged**